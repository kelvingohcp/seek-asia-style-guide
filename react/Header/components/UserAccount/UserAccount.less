@import (reference) "~seek-asia-style-guide/theme";

.secondaryNav {
  display: none;
  @media @desktop {
    display: grid;
    grid-auto-flow: column;
    align-items: center;
    grid-gap: @grid-base * 2;
  }
}

.userAcclink:focus {
  outline: none;
}

.signup {
  margin-right: @grid-base*2;
}

.signup:focus {
  outline: none;
}

.displayText {
  color: @saGrey3;
}
