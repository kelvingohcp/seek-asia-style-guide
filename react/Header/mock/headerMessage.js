export default {
  playStore: {
    title: 'Google Play Store',
    url: 'https://play.google.com/store/apps/details?id=com.jobsdb&utm_campaign=MobileNavigationMenuAppDownload&utm_source=MobileNavigationMenu&utm_medium=banner'
  },
  appStore: {
    title: 'App Store',
    url: 'https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432?utm_campaign=MobileNavigationMenuAppDownload&utm_source=MobileNavigationMenu&utm_medium=banner'
  },
  resumesAndDocuments: {
    title: 'Resumes and Documents',
    url: '/hk/en/MyJobsDB/resumes-and-documents'
  },
  applicationHistoryList: {
    title: 'Application History',
    url: '/hk/en/MyJobsDB/ApplictationHistoryList'
  },
  jobAlertsList: {
    title: 'Job Alerts',
    url: '/hk/en/MyJobsDB/JobAlertsList'
  },
  saveJobs: {
    title: 'Save Jobs',
    url: '/saveJobs.do'
  },
  jobsDBTitle: {
    title: 'My jobsDB',
    url: '/hk/en/MyJobsDB'
  },
  resources: {
    title: 'Resources',
    url: '/en-hk/career-recruitment-advice/'
  },
  careerInsights: {
    title: 'Career Insights',
    url: '/hk/en/career-insights'
  },
  profile: {
    title: 'View profile',
    url: '/hk/en/MyJobsDB/profile'
  },
  invitation: {
    title: 'Job invitations',
    url: '/hk/en/Invitations/InvitationList'
  },
  logout: {
    title: 'Log out',
    url: '/hk/en/login/jobseekerlogoff'
  },
  signUp: {
    title: 'Sign up',
    url: '/hk/en/jobseeker/registration'
  },
  login: {
    title: 'Log in',
    url: {
      home: '/hk/en/login/jobseekerlogin?from=header',
      jobListing: '/hk/en/login/jobseekerlogin?from=header&pageType=jobListing',
      jobAd: '/hk/en/login/jobseekerlogin?from=header&pageType=jobAd'
    }
  },
  employer: {
    title: 'For Employers',
    url: '/hk/en/home/employerindex'
  },
  homeTitle: 'Home',
  homeUrlText: 'jobsDB.com',
  homeUrl: '/hk',
  mobileLogoutUrl: '/logout.do',
  mobileLoginUrl: {
    home: '/login.do?nextPage=jobsForYou.do%3F',
    jobListing: '/login.do?nextPage=jobsForYou.do%3F&pageType=jobListing',
    jobAd: '/login.do?nextPage=jobsForYou.do%3F&pageType=jobAd'
  },
  downloadApp: 'Download Our App',
  seperatorOr: 'or',
  searchUrl: '/hk/en/search/getsearchresult'
};
