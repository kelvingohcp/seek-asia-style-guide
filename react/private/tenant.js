import Constants from '../Constants/Constants';
const { JOBSTREET, JOBSDB } = Constants;
export default tenant => ({
  isJobStreet: tenant === JOBSTREET,
  isJobsDB: tenant === JOBSDB
});
