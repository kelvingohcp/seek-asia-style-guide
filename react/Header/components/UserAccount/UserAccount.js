import React, { Component } from 'react';
import { Button, Text } from 'seek-asia-style-guide/react';
import PropTypes from 'prop-types';
import { AUTHENTICATED, UNAUTHENTICATED, AUTH_PENDING } from '../../../private/authStatusTypes';
import styles from './UserAccount.less';
import UserAccountMenu from '../UserAccountMenu/UserAccountMenu';
import _get from 'lodash/get';

export default class UserAccount extends Component {
  static defaultProps = {
    onMenuToggle: () => { },
    authenticationStatus: UNAUTHENTICATED
  }

  constructor(props) {
    super(props);

    this.state = {
      menuOpen: false
    };
  }

  handleMenuToggleClick = () => {
    this.setState(state => {
      this.props.onMenuToggle({ open: !state.menuOpen });
      return { menuOpen: !state.menuOpen };
    });
  };

  render() {
    const {
      loginAvailable,
      authenticationStatus,
      userName,
      userAccMenuItems,
      headerMessage,
      baseUrl,
      linkRenderer
    } = this.props;

    if (loginAvailable) {
      switch (authenticationStatus) {
        case AUTHENTICATED:
          return (
            <div className={styles.secondaryNav}>
              <span>
                {<UserAccountMenu userAccMenuItems={userAccMenuItems} userName={userName} linkRenderer={linkRenderer} />}
              </span>
            </div>
          );

        case UNAUTHENTICATED:
          return (
            <div className={styles.secondaryNav}>
              {linkRenderer({
                href: `${baseUrl + _get(headerMessage, 'login.url')}`,
                children: <Button
                  className={styles.userAcclink}
                  color='hyperlink'
                  data-automation='log-in'>
                  {`${_get(headerMessage, 'login.title')}`}
                </Button>
              })}
              <Text className={styles.displayText}>{_get(headerMessage, 'seperatorOr')}</Text>
              {linkRenderer({
                href: `${baseUrl + _get(headerMessage, 'signUp.url')}`,
                children: <Button
                  className={styles.signup}
                  color='hyperlink'
                  data-automation='sign-up'>
                  {`${_get(headerMessage, 'signUp.title')}`}
                </Button>
              })}
              {
                linkRenderer({
                  href: `${baseUrl + _get(headerMessage, 'employer.url')}`,
                  children: <Button
                    className={styles.userAcclink}
                    color='tertiary'
                    compact
                    data-automation='employer-site'>
                    {`${_get(headerMessage, 'employer.title')}`}
                  </Button>
                })}
            </div>
          );
        default:
          return null;
      }
    }
    return null;
  }
}

UserAccount.propTypes = {
  loginAvailable: PropTypes.bool.isRequired,
  authenticationStatus: PropTypes.oneOf([
    AUTHENTICATED,
    UNAUTHENTICATED,
    AUTH_PENDING
  ]).isRequired,
  userName: PropTypes.string,
  brandStyles: PropTypes.object,
  loginTitle: PropTypes.string,
  loginUrl: PropTypes.string,
  signupTitle: PropTypes.string,
  signupUrl: PropTypes.string,
  userAccMenuItems: PropTypes.array,
  onMenuToggle: PropTypes.func,
  baseUrl: PropTypes.string,
  headerMessage: PropTypes.object.isRequired,
  linkRenderer: PropTypes.func
};
