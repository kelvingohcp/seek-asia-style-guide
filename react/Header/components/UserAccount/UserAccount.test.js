import React from 'react';
import { shallow, render } from 'enzyme';
import headerMessage from '../../mock/headerMessage';
import { AUTHENTICATED, UNAUTHENTICATED } from '../../../private/authStatusTypes';

import UserAccount from './UserAccount';

describe('UserAccount', () => {
  const defaultProps = {
    headerMessage,
    authenticationStatus: AUTHENTICATED,
    loginAvailable: true,
    userName: 'JObsDB',
    userAccMenuItems: [],
    linkRenderer: props => (<a {...props} />),
    pageType: 'home'
  };

  it('should render with default props', () => {
    const wrapper = shallow(<UserAccount {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render is UNAUTHENTICATED', () => {
    const wrapper = shallow(<UserAccount {...{ ...defaultProps, authenticationStatus: UNAUTHENTICATED }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render login url based on pagetype', () => {
    const wrapper = render(<UserAccount {...{ ...defaultProps, authenticationStatus: UNAUTHENTICATED, pageType: 'jobListing' }} />);
    expect(wrapper).toMatchSnapshot();
    const wrapper2 = render(<UserAccount {...{ ...defaultProps, authenticationStatus: UNAUTHENTICATED, pageType: 'jobAd' }} />);
    expect(wrapper2).toMatchSnapshot();
  });
});

