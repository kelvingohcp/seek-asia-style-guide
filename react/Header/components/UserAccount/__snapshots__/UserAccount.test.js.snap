// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAccount should render is UNAUTHENTICATED 1`] = `
<div
  className="secondaryNav"
>
  <Text
    baseline={false}
    className="displayText"
  >
    or
  </Text>
</div>
`;

exports[`UserAccount should render with default props 1`] = `
<div
  className="secondaryNav"
>
  <span>
    <UserAccountMenu
      linkRenderer={[Function]}
      userAccMenuItems={Array []}
      userName="JObsDB"
    />
  </span>
</div>
`;
